### PR TITLE
Download CSV file of all service emails

### PR DESCRIPTION
**Pull request by georgeschena**
This PR allows a super admin to download all the organisations service emails in CSV format.
Screenshot 2019-05-13 at 12 30 59

### CODE REVIEW

File: app/controllers/admin/contact_all_orgs_controller.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Admin::ContactAllOrgsController < AdminController
```
Comment (tommotaylor) [inline app/controllers/admin/contact_all_orgs_controller.rb]: Not a very RESTful name.

File: app/models/organisation.rb [hunk 2]
```diff
@@ -22,6 +22,16 @@ def self.fetch_organisations_from_register
...
   end
 
+  def to_csv
+    attributes = %w{service_email}
+    CSV.generate(headers: true) do |csv|
+      csv << attributes
+      all.each do |location|
+        csv << location.attributes.values_at(*attributes)
+      end
+    end
```
Comment (tommotaylor) [inline app/models/organisation.rb]: Looks good but curious as to why you chose the model over the controller

File: config/application.rb [hunk 3]
```diff
@@ -1,6 +1,7 @@
 require_relative 'boot'
 
 require "rails"
+require 'csv'
```
Comment (tommotaylor) [inline config/application.rb]: Is this needed? `CSV` doesn't come with rails?

File: config/routes.rb [hunk 4]
```diff
@@ -69,6 +69,7 @@
 
   namespace :admin do
     resources :govwifi_map, only: %i[index]
+    resources :contact_all_orgs, only: %i[index]
```
Comment (tommotaylor) [inline config/routes.rb]: Not a very resourceful route

File: spec/features/super_admin/download_all_service_emails_spec.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
+describe 'Downloading all the service emails', type: :feature do
+  let(:user) { create(:user, :super_admin) }
```
Comment (tommotaylor) [inline spec/features/super_admin/download_all_service_emails_spec.rb]: The latest changes to the user factory means you need to explicitly pass in the trait for organisation, eg:
```
let(:user) { create(:user, :super_admin, ;with_organisation) }
```

File: spec/requests/organisations/get_spec.rb [hunk 6]
```diff
@@ -0,0 +1,17 @@
+describe "GET /locations", type: :request do
```
Comment (emileswarts) [inline spec/requests/organisations/get_spec.rb]: /locations old description

**Pull request merged**